addon/mixins/time-series-labeler.js

```diff
@@ -220,95 +255,86 @@ export default Ember.Mixin.create({
-        d = domainTypes[j];
-        results.push(this.numTimeBetween(domainTypeToLongDomainType[d], start, stop));
-      }
-      return results;
-    }).call(this);
-    labels = null;
     maxNumberOfLabels = this.get('maxNumberOfLabels');
+    maxNumberOfMinorTicks = this.get('maxNumberOfMinorTicks');
+    len = labelCandidates.length;
+
+    if (len > maxNumberOfLabels && typeof(this.customFilterLibrary[domain]) === "function") {
+      labelCandidates = this.customFilterLibrary[domain](maxNumberOfLabels, maxNumberOfMinorTicks, labelCandidates);
+      len = labelCandidates.length;
+    } else if (len > maxNumberOfLabels) {
+      // This tells us how many times we can half the results until we're at or
+      // below our maxNumberOfLabels threshold. Derived from:
+      // len ∕ 2ⁿ ≤ maxNumberOfLabels
+      // Math.log(x) / Math.LN2
+      modulo = Math.ceil(Math.log(len/(maxNumberOfLabels * (maxNumberOfMinorTicks + 1))) / Math.LN2) + 1;
+      labelCandidates = labelCandidates.filter(function(d, i) {
+        return i % Math.pow(2, modulo) === 0;
+      });
+      len = labelCandidates.length;
+    }
 
-    for (i = j = 0, len = times.length; j < len; i = ++j) {
-      timeBetween = times[i];
-
-      // quarter labels are calculated by simply getting month labels with 3
-      // month gaps
-      interval = null;
-      if (timeBetween < maxNumberOfLabels) {
-        interval = domainTypes[i] === 'Q' ? this.MONTHS_IN_QUARTER : 1;
-      } else if (domainTypes[i] === this.get('maxTimeSpecificity') || times[i + 1] < maxNumberOfLabels * (this.get('SPECIFICITY_RATIO'))) {
-        if (domainTypes[i] === 'Q') {
-          interval = Math.ceil(this.MONTHS_IN_QUARTER * timeBetween / maxNumberOfLabels);
-        } else {
-          interval = Math.ceil(timeBetween / maxNumberOfLabels);
-        }
-      }
-      if (interval != null) {
-        this.set('xAxisTimeInterval', domainTypes[i]);
-        labels = this.filterLabels(d3.time[labellerTypes[i]](start, stop), interval);
-        break;
-      }
+    // So now we figure out (if we have added space for) the number of minor
+    // ticks that will be shown in the presentiation.
+    if (maxNumberOfMinorTicks > 0) {
+      this.set('minorTickInterval', Math.ceil(len / maxNumberOfLabels));
     }
-    return labels;
+    return labelCandidates;
   },
-  filterLabels: function(array, interval){
-    return array.filter(function filterLabels(d, i) {
-      return i % interval === 0;
+
+  filterLabelsForQuarters: function(dates){
+    // Pretty simple; getMonth is a 0 based index of the month.  We do modulo
+    // for the time being.
+    return dates.filter(function(d) {
+      return d.getMonth() % 3 === 0;
     });
   },
+
+  // We have an option of suppling custom filters based on the date type.  This
+  // way we can append any special behavior or pruning algorythim to Months that
+  // wouldn't be applicable to Weeks
+  customFilterLibrary: {},
```

Comment:
Actually this refers to a story, an _Allegory_rhythm  ;-)
....
Okay... I can't spell worth a damn T_T